### PR TITLE
#849 unit test and solution

### PR DESCRIPTION
**Pull request by sjaakd**

### CODE REVIEW

File: processor/src/test/java/org/mapstruct/ap/test/bugs/_849/Issue849Test.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mapstruct.ap.testutil.IssueKey;
+import org.mapstruct.ap.testutil.WithClasses;
+import org.mapstruct.ap.testutil.runner.AnnotationProcessorTestRunner;
+
+/**
+ * @author Andreas Gudian
```
Comment (agudian) [inline processor/src/test/java/org/mapstruct/ap/test/bugs/_849/Issue849Test.java]: who's that?

File: processor/src/test/java/org/mapstruct/ap/test/bugs/_849/Issue849Test.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+ *
+ */
+@WithClasses({ Source.class, Target.class, Issue849Mapper.class })
+@RunWith(AnnotationProcessorTestRunner.class)
+public class Issue849Test {
+
+    @Test
+    @IssueKey("849")
+    public void shouldApplyCorrectOrderingWithDependsOn() {
```
Comment (agudian) [inline processor/src/test/java/org/mapstruct/ap/test/bugs/_849/Issue849Test.java]: `shouldCompileWithAllImportsDeclared`?

File: processor/src/test/java/org/mapstruct/ap/test/bugs/_849/Source.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+import java.io.Serializable;
+import java.util.List;
+
+/**
+ *
+ * @author Sjaak Derksen
+ */
+public class Source {
+  private List<Serializable> sourceList;
```
Comment (agudian) [inline processor/src/test/java/org/mapstruct/ap/test/bugs/_849/Source.java]: The indentation in `Source` and `Target` is off. Half off maybe. :wink:

**Comment by agudian** [comment]
Just have some small stuff nag about in the tests. If you could also reword the commit message a bit more, that'd be great. Other than that, feel free to push to master! :+1:

**Comment by sjaakd** [comment]
please test this

**Comment by sjaakd** [comment]
test this please

**Pull request merged**